src/main/kotlin/com/sourcegraph/cody/agent/protocol/util/Rfc3986UriEncoder.kt

```diff
@@ -0,0 +1,15 @@
+package com.sourcegraph.cody.agent.protocol.util
+
+object Rfc3986UriEncoder {
+
+  // todo solve this with library
```

Comment:
have you found any libs that could do that?

Reply:
Yeah, that's why I've added `todo`. Libs I've checked so far:
* Apache HTTP client encoder - comma not encoded, partition not normalized
* Spring Web - similar, but also version I've checked has some CVEs
* https://github.com/dmfs/uri-toolkit - I haven't checked that, but development is not active